### PR TITLE
Curate sources to avoid the UI crashing

### PR DESCRIPTION
**Pull request by imartinez**
Remove sources from chat history to avoid confusing the LLM

Fixes https://github.com/imartinez/privateGPT/issues/1189

### CODE REVIEW

File: private_gpt/ui/ui.py [hunk 1]
```diff
@@ -32,6 +42,23 @@ def __init__(self) -> None:
...
+                and "file_name" in chunk.document.doc_metadata
+                else "-",
+                page=chunk.document.doc_metadata["page_label"]
+                if chunk.document.doc_metadata
+                and "page_label" in chunk.document.doc_metadata
+                else "-",
+                text=chunk.text,
+            )
+            for chunk in sources
+        }
```
Comment (lopagela) [inline private_gpt/ui/ui.py:60]: Could have been in `Source` class

```python
# Usage would look like
Source.curated_from_chunks(sources)
```

Comment (lopagela) [reply private_gpt/ui/ui.py:60]: There are two identical check made on `bool(chunk.document.doc_metadata)` - could be reduced to one

Additionally, accessing to `page_label` and `file_name` can still cause key errors (the calls to the dict should be made with `.get(key, _default_value)`

File: private_gpt/ui/ui.py [hunk 2]
```diff
@@ -9,18 +9,28 @@
...
 
-
 UI_TAB_TITLE = "My Private GPT"
+SOURCES_SEPARATOR = "\n\n Sources: \n"
+
+
+class Source(BaseModel):
+    file: str
+    page: str
+    text: str
```
Comment (lopagela) [inline private_gpt/ui/ui.py:30]: Nice idea to have that dedicated object, it reduce the code dup around metadata management ✌️

**Review by lopagela — COMMENTED** [review]
I believe the key not found error is still present

File: private_gpt/ui/ui.py (hunk 1 above)
Comment (imartinez) [reply private_gpt/ui/ui.py]: Moved and rewrote the whole method addressing these two comments. Thanks!

**Review by pabloogc — APPROVED** [review]

**Pull request merged**